postgresql/installingpostgresqlonibmi.md

```diff
@@ -1,54 +1,87 @@
-# Install and configure Postgresql DB Server on IBM i 
+# PostgreSQL Installation and Basic Configuration on IBM i
 
-Install postgres yum packages from IBM i ACS Open Source Package Management
-```
-Package list:
-postgresql12 ***(This item may be the only one you need to select to include all packages)***
-postgresql12-contrib
-postgresql12-devel
-postgresql12-docs
-postgresql12-libecpg
-postgresql12-libpgtypes
-postgresql12-libpq
-postgresql12-server
+The purpose of this document is to summarize PostgreSQL installation and basic configuration on IBM i. The following topics are outlined below: initial setup, enabling remote connections, server startup, and connecting with a client.
+
+## Initital Setup
+
+Install the `postgres12-server` and `postgresql-contrib` rpm packages using the [IBM i ACS Open Source Package Management](https://www.ibm.com/support/pages/node/706903) wizard or an SSH shell session with:
+:w
+```bash
+$ yum install postgresql12-server postgresql12-contrib
 ```
 
-Create IBM i user: ***POSTGRES*** and give it a password. It's important the user is named POSTGRES because the Postgresql server needs a default user of POSTGRES. At least that was my experience with it. (Also give it *ALLOBJ access for now. You can probably remove later and possibly even disable the IBM i POSTGRES user. ***Not tested.***)
+According to the [Postgresql docs](https://www.postgresql.org/docs/12/postgres-user.html), "It is advisable to run PostgreSQL under a separate user account. This user account should only own the data that is managed by the server, and should not be shared with other daemons. The user name postgres is often used but you can use another name if you like."
 
-Log in to SSH as IBM i user ***POSTGRES*** to do server setup.
+From a 5250 session create a ***POSTGRES*** user profile with:
+
+```
+CRTUSRPRF  USRPRF(POSTGRES)  PASSWORD(...) HOMEDIR('/postgres')
+```
 
-If not started, start the bash shell by typing ***bash*** unless bash is already your default shell. 
+Next create the `/postgres` directory and change ownership to the ***POSTGRES*** user. From SSH or QSH session run:
 
-Run the following shell command line sequence to initialize postgres database in the /postgres IFS directory location
+```sh
+$ mkdir /postgres
+$ chown postgres /postgres
 ```
-cd /
 
-mkdir postgres
+***NOTE:*** Creating a directory under `/` requires a user with `*ALLOBJ` authority.
 
-export PGDATA=/postgres
+Log in to your IBM i via SSH as the **POSTGRES** user.
 
-initdb -E UTF-8 -D /postgres
+If not started, start the bash shell by typing ***bash*** unless bash is already your default shell.
+
+Run the following commands to initialize PostgreSQL database in the `/postgres` IFS directory location
+
+```bash
+
+$ export PGDATA=/postgres
+
+$ initdb -E UTF-8 -D /postgres -W -A scram-sha-256
 ```
 
-After initdb you should see the following message:
+You will be prompted to `Enter new superuser password` for the **POSTGRES** user.
```

Comment:
We should note that this password is distinct from the password for the user profile.